pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -588,36 +558,7 @@ func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstID string, ap
 		reportAppInstMetadata.Amdinfo = ReportAppInstMetaData
 	}
 
-	log.Functionf("PublishAppInstMetaDataToZedCloud sending %v", ReportInfo)
-
-	data, err := proto.Marshal(ReportInfo)
-	if err != nil {
-		log.Fatal("PublishAppInstMetaDataToZedCloud proto marshaling error: ", err)
-	}
-	statusURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "info")
-
-	deferKey := "appInstMetadataInfo:" + appInstID
-	zedcloud.RemoveDeferred(zedcloudCtx, deferKey)
-
-	buf := bytes.NewBuffer(data)
-	if buf == nil {
-		log.Fatal("malloc error")
-	}
-	size := int64(proto.Size(ReportInfo))
-	start := time.Now()
-	err = SendProtobuf(statusURL, buf, size, iteration)
-	if err != nil {
-		log.Errorf("PublishAppInstMetaDataToZedCloud failed: %s", err)
-		// Try sending later
-		// The buf might have been consumed
-		buf := bytes.NewBuffer(data)
-		if buf == nil {
-			log.Fatal("malloc error")
-		}
-		zedcloud.SetDeferred(zedcloudCtx, deferKey, buf, size, statusURL, true)
-	} else {
-		log.Functionf("sent appInstMetadata for appInstID: %v, took: %v", appInstID, time.Since(start))
-	}
+	schedulePublishMessageToZedCloud(ctx, ReportInfo, appInstID)
```

Comment:
You musr use the deferKey in the origin code from line 599

Reply:
I split all objects by type, so, I think it is not needed